src/Shared/Components/WebAssemblyComponentMarker.cs

```diff
@@ -9,20 +11,20 @@ internal struct WebAssemblyComponentMarker
     {
         public const string ClientMarkerType = "webassembly";
 
-        public WebAssemblyComponentMarker(string type, string assembly, string typeName, string parameterDefinitions, string parameterValues, string prereenderId) =>
-            (Type, Assembly, TypeName, ParameterDefinitions, ParameterValues, PrerenderId) = (type, assembly, typeName, parameterDefinitions, parameterValues, prereenderId);
+        public WebAssemblyComponentMarker(string? type, string? assembly, string? typeName, string? parameterDefinitions, string? parameterValues, string? prerenderId) =>
+            (Type, Assembly, TypeName, ParameterDefinitions, ParameterValues, PrerenderId) = (type, assembly, typeName, parameterDefinitions, parameterValues, prerenderId);
 
-        public string Type { get; set; }
+        public string? Type { get; set; }
```

Comment:
This isn't the place for this but since we're going full-on with nullability, we should consider making the end record marker its own type. It feels weird to reduce the nullability requirements on a type for a single instance of it.

Reply:
That's a fair assessment. Let me see if I can squeeze that change here